app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt

```diff
@@ -0,0 +1,150 @@
+package com.github.se.eventradar.viewmodel
+
+import androidx.lifecycle.ViewModel
+import com.github.se.eventradar.model.User
+import com.github.se.eventradar.model.message.MessageHistory
+import com.github.se.eventradar.model.repository.message.IMessageRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+
+@HiltViewModel
+class ChatViewModel
+@Inject
+constructor(
+    private val messagesRepository: IMessageRepository,
+    private val userRepository: IUserRepository,
+) : ViewModel() {
+  private val _uiState = MutableStateFlow(ChatUiState())
+  val uiState: StateFlow<ChatUiState> = _uiState
+
+  fun onMessageBarInputChange(newInput: String) {
+    _uiState.update { currentState -> currentState.copy(messageBarInput = newInput) }
+  }
+
+  // TO DO: Complete VM functions for chat screen
```

Comment:
The commented code below will be addressed when the VM is implemented in a future PR. To maintain a clean codebase, I suggest removing this unused code now.

Reply:
Agree, deleted.